owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/security/PatternViewModel.kt

```diff
@@ -42,4 +49,23 @@ class PatternViewModel(
         val savedPattern = preferencesProvider.getString(PatternActivity.PREFERENCE_PATTERN, null)
         return savedPattern != null && savedPattern == patternValue
     }
+
+    fun isBiometricLockAvailable(): Boolean {
+        return if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            false
+        } else if (!BiometricManager.isHardwareDetected()) { // Biometric not supported
+            false
+        } else BiometricManager.hasEnrolledBiometric() // Biometric not enrolled
+    }
+
+    private fun getPreferenceManagerEditor(): SharedPreferences.Editor {
+        val appPrefsEditor = PreferenceManager.getDefaultSharedPreferences(contextProvider.getContext())
+        return appPrefsEditor.edit()
+    }
+
+    fun setBiometricsState(enabled: Boolean) {
+        val editor = getPreferenceManagerEditor()
+        editor.putBoolean(BiometricActivity.PREFERENCE_SET_BIOMETRIC, enabled)
+        editor.apply()
+    }
```

Comment:
Here same comments as in `PassCodeViewModel`

Reply:
Done